src/core_plugins/tile_map/public/maps_visualization.js

```diff
@@ -261,10 +272,47 @@ export function MapsVisualizationProvider(serviceSettings, Notifier, getAppState
       this.vis.updateState();
     }
 
-  }
+    async getGeohashBounds() {
```

Comment:
Not sure about this.

Each layer should determine its bounds. We're throwing that away when creating this function that needs to be supplied to the FitControl.

I'd move this method to `geohash_bounds`, and modify `fitToData` so it can work with async results.